design/ceph-deprecate-old-osd-types.md

```diff
@@ -0,0 +1,433 @@
+the cluster's OSDMap should be started.
+
+Rook may opt to delete OSDs which have been removed from the Ceph cluster config and have been `out`
+for longer than some threshold period measured in days, e.g., 45 days. Rook will occasionally report
+in its logs (at least once per day) that it intends to delete an OSD after some number of days.
+
+####### PVCs as backing storage
+In this mode, users will also be able to specify Kubernetes Persistent Volume Claims to use as
+backing storage for Rook. This is a commonly-requested feature that is planned for addition into
+Rook v1.1. This plan must therefore account for this.
+
+##### Ceph-managed mode
+This mode is characterized by both of the Ceph cluster configuration options `useAllDevices: false`,
+`useAllHosts: false`, `deviceFilter: <unset>`, and `devices: <unset>`. Any device that is found on
+any host available to Rook that is labeled with the Ceph cluster UUID and corresponds to a valid OSD
+in the cluster's OSDMap should be started. New OSDs could be created manually by a human operator or
+by Ceph's Rook orchestrator. The Rook orchestrator initiates device creation by triggering a
+Kubernetes job that runs a `ceph-volume` prepare sequence, at which point the Rook operator's
+control loop will discover a valid device and start the OSD.
+
+#### Future security concerns
+While planning for future extensibility of OSDs, a notable future development will be attempts to
+reduce the privilege requirements and system access which Rook's OSD-related pods will have. While
+the specific details of how this will be achieved are out of scope for this design, some details are
+known currently which can help influence this design, making it flexible for future developments.
+
+The most critical design choice which can help reduce pod permissions is to give pods access to only
+the host storage that is required for a given operation. This may mean mounting only certain paths
+and/or devices into pods, and it may mean mounting paths/devices in read-only mode when possible.
+
+Reasonable attempts to secure Rook during execution of this design can be made, but it is not in the
+scope of this design that many hours be spent focusing on security initially. Where security is
+mentioned in this design, a reasonable product-of-opportunity attempt may be possible, but
+completion of this design is not predicated on completion of security concerns.
+
+#### Test environments without disks
+Rook must occasionally support test environments without disks. Users new to Rook especially may not
+want to create block devices for their initial test environments. Ceph Bluestore has the ability to
+use files as disks for such test environments, and this design proposes to enable this as an option
+for such environments.
+
+To enable this feature, the Rook-Ceph operator will observe the environment variable
+`TEST_ENVIRONMENT_FILE_DEVICE_SIZE_GiB`. If this variable is set with a positive, nonzero value, the
```

Comment:
Bluestore block size default is 10GB, this seems reasonable for developers so I would not make this configurable. I'd like to highlight one of your earlier comments @BlaineEXE, the target audience for this must be developers only, no one should be encouraged to use this in production, keeping the default 10GB block size is one way.

Thus I'd turn this variable to something like: `TEST_ENVIRONMENT_OSD_ON_FILE`.

Reply:
Per my discussion with @invidian, some users may still wish to use this feature occasionally, and some flexibility to set the size would be desired. 

We could assume to round up to the nearest 10GB within the operator to better handle the 10GB block size.

When you say 10GB, do you mean Gigabytes or Gibibytes?